cdm/core/src/main/java/ucar/nc2/NetcdfFiles.java

```diff
@@ -290,31 +364,40 @@ else if (uriString.startsWith("file:"))
 
   private static ucar.unidata.io.RandomAccessFile getRaf(String location, int buffer_size) throws IOException {
     String uriString = location.trim();
-
     if (buffer_size <= 0)
       buffer_size = default_buffersize;
 
-    ucar.unidata.io.RandomAccessFile raf;
-    if (uriString.startsWith("http:") || uriString.startsWith("https:")) { // open through URL
-      raf = new ucar.unidata.io.http.HTTPRandomAccessFile(uriString);
-
-    } else if (uriString.startsWith("nodods:")) { // deprecated use httpserver
-      uriString = "http" + uriString.substring(6);
-      raf = new ucar.unidata.io.http.HTTPRandomAccessFile(uriString);
-
-    } else if (uriString.startsWith("httpserver:")) { // open through URL
-      uriString = "http" + uriString.substring(10);
-      raf = new ucar.unidata.io.http.HTTPRandomAccessFile(uriString);
-
-    } else if (uriString.startsWith("slurp:")) { // open through URL
-      uriString = "http" + uriString.substring(5);
-      byte[] contents = IO.readURLContentsToByteArray(uriString); // read all into memory
-      raf = new InMemoryRandomAccessFile(uriString, contents);
-    } else if (uriString.startsWith("s3:")) {
-      raf = new S3RandomAccessFile(uriString);
-      if (isCompressed(uriString))
-        raf = downloadAndDecompress(raf, uriString, buffer_size);
-    } else {
+    ucar.unidata.io.RandomAccessFile raf = null;
+
+    for (RandomAccessFileProvider provider : registeredRandomAccessFileProviders) {
+      if (provider.isOwnerOf("location")) {
```

Comment:
is this a typo?

if (provider.isOwnerOf("location"))

Reply:
Yep, total typo. Good catch.